clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -0,0 +1,359 @@
+      Map<Integer, List<String>> routingInfo =
+          ((HashMap<Utf8, Collection<Utf8>>) metadataResponse.get("routingInfo")).entrySet()
+              .stream()
+              .collect(
+                  Collectors.toMap(
+                      e -> Integer.valueOf(e.getKey().toString()),
+                      e -> e.getValue().stream().map(Utf8::toString).collect(Collectors.toList())));
+
+      if (fetchedVersion != getCurrentStoreVersion()) {
+        // call the DICTIONARY endpoint if needed
+        CompletableFuture<TransportClientResponse> dictionaryFetchFuture = null;
+        if (!versionZstdDictionaryMap.containsKey(fetchedVersion)
+            && compressionStrategy == CompressionStrategy.ZSTD_WITH_DICT.getValue()) {
+          dictionaryFetchFuture = fetchCompressionDictionary(fetchedVersion);
+        }
+
+        // Update partitioner pair map (versionPartitionerMap)
+        Properties params = new Properties();
+        params.putAll(partitionerParams);
+        VenicePartitioner partitioner =
+            PartitionUtils.getVenicePartitioner(partitionerClass, amplificationFactor, new VeniceProperties(params));
+        versionPartitionerMap.put(fetchedVersion, new PartitionerPair(partitioner, partitionCount));
+
+        // Update readyToServeInstanceMap
+        for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+          String key = getVersionPartitionMapKey(fetchedVersion, partitionId);
+          readyToServeInstancesMap.put(key, routingInfo.get(partitionId));
+        }
+
+        // Update schemas
+        SchemaData schemaData = new SchemaData(storeName);
+        for (Map.Entry<Integer, String> entry: keySchema.entrySet()) {
+          schemaData.setKeySchema(new SchemaEntry(entry.getKey(), entry.getValue()));
+        }
+        for (Map.Entry<Integer, String> entry: valueSchemas.entrySet()) {
+          schemaData.addValueSchema(new SchemaEntry(entry.getKey(), entry.getValue()));
+        }
+        schemas.set(schemaData);
+
+        // Evict entries that are over two versions behind from the latest fetched version
+        readyToServeInstancesMap.entrySet().removeIf(entry -> getVersionFromKey(entry.getKey()) < fetchedVersion - 2);
+        versionPartitionerMap.entrySet().removeIf(entry -> entry.getKey() < fetchedVersion - 2);
+        versionZstdDictionaryMap.entrySet().removeIf(entry -> entry.getKey() < fetchedVersion - 2);
+
+        // Wait for dictionary fetch to finish if there is one
+        try {
+          if (dictionaryFetchFuture != null) {
+            dictionaryFetchFuture.get(ZSTD_DICT_FETCH_TIMEOUT, TimeUnit.SECONDS);
+          }
+          currentVersion.set(fetchedVersion);
+          clusterStats.updateCurrentVersion(getCurrentStoreVersion());
+          latestSuperSetValueSchemaId.set(newSuperSetValueSchemaId);
+        } catch (InterruptedException interruptedException) {
+          Thread.currentThread().interrupt();
```

Comment:
Do we need this `catch` block? If we're keeping this `catch` block does the function still need `throws InterruptedException` ?

Reply:
Went ahead and removed, I removed the `catch` on the out `try` for metadata fetch but just forgot to remove the inner one for dictionary fetch